extending/provider/implementation.rst

```diff
@@ -1,85 +1,44 @@
 Implementation
 --------------
 
-Annotations
-+++++++++++
+Every cloud provider is made up of one or more resources. Each resource implementation will manage the lifecycle
+of a resource in the cloud: create, read, update, and delete.
 
-**@Type(string)**
-    The class should be annotated with the ``@Type(string)`` annotation. The name provided by this annotation is
-    used by the Gyro language to lookup the resource implementation. For example, ``@Type("instance")`` in the AWS
-    provider will make ``aws::instance`` available to the Gyro language.
+A resource is any entity in a cloud provider that has state. For example, ``aws::load-balancer`` or
+``aws::instance`` are resources with state where-as a service like AWS Rekognition has no persistent state so
+it would not need to be implemented as a resource in Gyro.
 
-    The Java package(s) that make up a provider should be annotated with one or more of the following:
+With the exception of ``refresh()`` all CRUD methods provide a ``GyroUI`` and ``State`` objects.
 
-**@Namespace(string)**
-    This is name the primary namespace of the provider that will be exposed to the Gyro language. For example, for
-    the resource ``aws::instance`` this annotation defines the ``aws`` portion of the resource name. The namespace
-    should be short, all-lower case, unique, and concisely describe the provider.
+The ``GyroUI`` objects can be used to output additional information to the user's console.
 
-    This is a ``package`` annotation intended to be used in a ``package-info.java`` file at the root of the provider
-    projects main package.
+The ``State`` object can be used to periodically save state between operations within one of the CRUD methods. For
```

Comment:
You may want to first add a brief description on what the main purpose Gyro uses the ``State`` object.  Something like "Gyro uses the ``State`` object to maintain a persistent record of the values configured and many times values (read-only) returned by the provider. But it can also be saved during a create/update call... "